### PR TITLE
Script to build and package Cloud Tools for PowerShell

### PR DESCRIPTION
**Pull request by chrsmith**
Adds a script to automate the process of building and packaging the cmdlets so they can be
added to the Cloud SDK installer.

The script is very straight forward, the only tricky part is the introduction of a new
`GoogleCloudPowerShell.psd1` module. This is necessary to maintain backwards compatibility
with existing Cloud SDK installs due to the new module layout from the fix to #314 / #315.

Existing installs have the `PSModulePath` pointed to a `GoogleCloudPowerShell` folder that
will no longer exist after `gcloud components update` gets run. So we put this new module
in that location, so it will load the updated assemblies.

New installs will have their `PSModulePath` pointed to a `PowerShell` folder, and and will not
pick up the new module located in `..\GoogleCloudPowerShell` directory.

### CODE REVIEW

**Comment by chrsmith** [comment]
@ILMTitan PTAL

File: Tools/BuildAndPackage.ps1 [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Builds the Google Cloud Tools for PowerShell project, and packages the output so it can be
+# integrated into the Google Cloud SDK. This is really of only use to Googlers doing the
+# official releases of the PowerShell module.
+
+# Bin directory for the Google.PowerShell project.
+$binDir = Join-Path $PSScriptRoot "\..\Google.PowerShell\bin\"
+$binDir = [System.IO.Path]::GetFullPath($binDir)
```
Comment (ILMTitan) [inline Tools/BuildAndPackage.ps1]: `$binDir = Resolve-Path $binDir` perhaps?

File: Tools/BuildAndPackage.ps1 [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+Write-Host -ForegroundColor Cyan "`n*** Purging the bindir***"
+if (Test-Path $binDir) {
+    Remove-Item $binDir -Recurse -Force
+}
+Write-Host "done`n"
+
+# Build the project.
+Write-Host -ForegroundColor Cyan "`n*** Building the project ***"
+
+# TODO(chrsmith): Enable the build from this script. The project's PostBuild step requires some
```
Comment (ILMTitan) [inline Tools/BuildAndPackage.ps1]: This worked fine on my computer. Which version of MSBuild were you using?

File: Tools/BuildAndPackage.ps1 [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+# HACK: Move the GoogleCloudPowerShell.psd1 into a different folder to keep Cloud SDK installations
+# before 9/29/2016 working. We changed the location of the path we add to PSModulePath during the
+# Cloud SDK installation. The old path (<CloudSDK>\platform\GoogleCloudPowerShell) will no longer
+# exist, since the module is put in (<CloudSDK>\platform\PowerShell\GoogleCloud). We drop a module
+# file in that location so that existing machines, with the old PSModulePath, will get updates.
+# TODO(chrsmith): Given that only ~2 months of Cloud SDK installs are effected, and those machines
+# may get paved, etc., and full uninstall/reinstall will fix it, consider removing this in H2 2017.
+Write-Host -ForegroundColor Cyan "`n*** HACK: Creating GoogleCloudPowerShell module ***"
+Write-Host "done`n"
```
Comment (ILMTitan) [inline Tools/BuildAndPackage.ps1]: Say done after moving the file?

File: Tools/BuildAndPackage.ps1 (hunk 1 above)
Comment (chrsmith) [inline Tools/BuildAndPackage.ps1]: `Resolve-Path` seems to require the path actually exist. So if the `$binDir` has already been deleted, `Resolve-Path` will fail.

**Comment by chrsmith** [comment]
PTAL

File: Tools/BuildAndPackage.ps1 [hunk 4]
```diff
@@ -2,46 +2,38 @@
...
-*******************
-"@
+Write-Host -ForegroundColor Cyan "*** Building the project ***"
+
+$msbuild = "c:\Program Files (x86)\MSBuild\14.0\Bin\MSBuild.exe"
+& $msbuild @($slnFile, "/t:Clean,Build", "/p:Configuration=Debug")
+
 if (-Not (Test-Path $debugDir)) {
-    Write-Host -ForegroundColor Red "ERROR: new binaries not found"
+    Write-Host -ForegroundColor Red "ERROR: Build results not found."
```
Comment (ILMTitan) [inline Tools/BuildAndPackage.ps1]: Why not `Write-Error`?

File: Tools/BuildAndPackage.ps1 [hunk 5]
```diff
@@ -69,24 +61,23 @@ $moduleDir = Join-Path $powerShellDir "Debug"
...
 
 # Ensure key files are in the right place.
-Write-Host -ForegroundColor Cyan "`n*** Sanity checking ***"
+Write-Host -ForegroundColor Cyan "*** Sanity checking ***"
 function ConfirmExists($relativePath) {
-   $fullPath = Join-Path $binDir $relativePath
+   $fullPath = Join-Path $packageDir $relativePath
    if (-Not (Test-Path $fullPath)) {
-       Write-Host -ForegroundColor Red "ERROR: Expected file $fullPath does not exist."
+       Write-Host -ForegroundColor Red "ERROR: Expected file '$fullPath' does not exist."
```
Comment (ILMTitan) [inline Tools/BuildAndPackage.ps1]: This could also be `Write-Error`

Comment (chrsmith) [inline Tools/BuildAndPackage.ps1]: Done.

**Comment by chrsmith** [comment]
Done.

**Pull request merged**